cooltools/snipping.py

```diff
@@ -481,4 +643,209 @@ def snip(self, exp, region1, region2, tup):
+    flank : int
+        How much to flank the center of the features by, in bp
+    min_diag: str or int
+        All diagonals of the matrix below this value are ignored. 'auto'
+        tries to extract the value used during the matrix balancing,
+        if it fails defaults to 2
+    clr_weight_name : str
+        Value of the column that contains the balancing weights
+    force : bool
+        Allows start>end in the features (not implemented)
+    nproc : str
+        How many cores to use
+    Returns
+    -------
+        np.ndarray: a stackup of all snippets corresponding to the features
+
+    """
+
+    if {"chrom", "start", "end"}.issubset(features_df.columns):
+        feature_type = "bed"
+    elif {"chrom1", "start1", "end1", "chrom2", "start2", "end1"}.issubset(
+        features_df.columns
+    ):
+        feature_type = "bedpe"
+    else:
+        raise ValueError("Unknown feature_df format")
+
+    features_df = assign_regions(features_df, view_df)
+
+    # Draft using bioframe assign_view, that doesn't work because of mysterious dtype issues!
+    # And with all the required renaming, it's more annoying than it could be...
+    # if feature_type == "bed":
+    #     features_df = bioframe.assign_view(features_df, view_df, df_view_col="region")
+    # else:
+    #     features_df = bioframe.assign_view(
+    #         features_df.rename(
+    #             columns={"chrom1": "chrom", "start1": "start", "end1": "end"}
+    #         ),
+    #         view_df,
+    #         df_view_col="region1",
+    #     ).rename(columns={"chrom": "chrom1", "start": "start1", "end": "end1"})
+    #     features_df = bioframe.assign_view(
+    #         features_df.rename(
+    #             columns={"chrom2": "chrom", "start2": "start", "end2": "end"}
+    #         ),
+    #         view_df,
+    #         df_view_col="region2",
+    #     ).rename(columns={"chrom": "chrom2", "start": "start2", "end": "end2"})
+    #     features["region"] = np.where(
+    #         features["region1"] == features["region2"], features["region1"], None
+    #     )
+    #     features = features.drop(["region1", "region2"], axis=1)
+
+    if flank is not None:
+        features_df = expand_align_features(
+            features_df, flank, clr.binsize, format=feature_type
+        )
+    else:
+        features_df = features_df.copy()
+        if feature_type == "bed":
+            features_df["lo"] = (features_df["start"] / clr.binsize).astype(int)
+            features_df["hi"] = (features_df["end"] / clr.binsize).astype(int)
+        else:
+            features_df["lo1"] = (features_df["start1"] / clr.binsize).astype(int)
+            features_df["hi1"] = (features_df["end1"] / clr.binsize).astype(int)
+            features_df["lo2"] = (features_df["start2"] / clr.binsize).astype(int)
+            features_df["hi2"] = (features_df["end2"] / clr.binsize).astype(int)
+
+    if view_df is None:
+        view_df = bioframe.make_viewframe(clr.chromsizes)
```

Comment:
i'd say use `make_cooler_view` helper function - mostly for consitency with other tools
but also because make_viewframe(chromsizes) yields a view, where names are UCSC by default, which isn't so nice when working on full chromosomzes